### PR TITLE
support a simplify merge cluase

### PR DESCRIPTION
**Pull request by swilly22**

### CODE REVIEW

File: src/execution_plan/ops/op_merge.c [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        op->edges = NULL;
+    }
+}
+
+OpBase* NewMergeOp(RedisModuleCtx *ctx, AST_Query *ast, Graph *g, QueryGraph *qg, const char *graph_name, ResultSet *result_set) {
+    return (OpBase*)NewMerge(ctx, ast, g, qg, graph_name, result_set);
+}
+
+OpMerge* NewMerge(RedisModuleCtx *ctx, AST_Query *ast, Graph *g, QueryGraph *qg, const char *graph_name, ResultSet *result_set) {
+    OpMerge *op_merge = calloc(1, sizeof(OpMerge));
```
Comment (jeffreylovitz) [inline src/execution_plan/ops/op_merge.c]: We can just malloc here - `nodes` and `edges` aren't initialized, but those are set before being accessed in `_SaveQueryGraphEntities`.

File: src/execution_plan/ops/op_merge.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#define __OP_MERGE_H
+
+#include "op.h"
+#include "../../graph/node.h"
+#include "../../graph/edge.h"
+#include "../../parser/ast.h"
+#include "../../resultset/resultset.h"
+
+/* Merge execution plan operation,
+ * this operation will create a pattern P if it doesn't not exists
```
Comment (jeffreylovitz) [inline src/execution_plan/ops/op_merge.h]: Bit of a typo!

File: src/resultset/resultset_statistics.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+#include <stdbool.h>
+
+typedef struct {
+    int labels_added;           /* Number of labels added as part of a create query. */
+    int nodes_created;          /* Number of nodes created as part of a create query. */
+    int properties_set;         /* Number of properties created as part of a create query. */
+    int relationships_created;  /* Number of edges created as part of a create query. */
+    int nodes_deleted;          /* Number of nodes removed as part of a delete query.*/
+    int relationships_deleted;  /* Number of edges removed as part of a delete query.*/
+} ResultSetStatistics;
```
Comment (jeffreylovitz) [inline src/resultset/resultset_statistics.h]: As an aside, I really like this change!

File: src/parser/grammar.y [hunk 4]
```diff
@@ -36,27 +36,31 @@
...
-	A = New_AST_Query(B, C, NULL, D, NULL, E, F, G);
+	A = New_AST_Query(B, C, NULL, NULL, D, NULL, E, F, G);
 }
 
 expr(A) ::= createClause(B). {
-	A = New_AST_Query(NULL, NULL, B, NULL, NULL, NULL, NULL, NULL);
+	A = New_AST_Query(NULL, NULL, B, NULL, NULL, NULL, NULL, NULL, NULL);
+}
+
+expr(A) ::= mergeClause(B). {
```
Comment (jeffreylovitz) [inline src/parser/grammar.y]: I had thought that merge syntax expected a RETURN clause, though I may have misunderstood.

Comment (swilly22) [reply src/parser/grammar.y]: In the case of MERGE, RETURN is not mandatory, and in our simplified version, where you can't create more than one pattern RETURN is not that important, at least from might perspective, although it shouldn't be too difficult to have a RETURN clause in addition to MERGE.

File: src/resultset/resultset_statistics.h (hunk 3 above)
Comment (swilly22) [reply src/resultset/resultset_statistics.h]: Thanks!

**Pull request merged**